src/Nethermind/Nethermind.Network/iTrustedNodesManager.cs

```diff
@@ -0,0 +1,18 @@
+// SPDX-FileCopyrightText: 2024 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Nethermind.Config;
+
+namespace Nethermind.Network
+{
+    public interface ITrustedNodesManager : INodeSource
+    {
+        IEnumerable<NetworkNode> Nodes { get; }
+        Task InitAsync();
+        Task<bool> AddAsync(string enode, bool updateFile = true);
+        Task<bool> RemoveAsync(string enode, bool updateFile = true);
+        bool IsTrusted(string enode);
```

Comment:
Not sure if we should have the interface based on strings? Maybe we should base it on `PublicKeys` or `Enode`?